packages/rrweb/src/record/observer.ts

```diff
@@ -365,15 +365,18 @@ function initInputObserver({
     ) {
       return;
     }
+
     let text = (target as HTMLInputElement).value;
     let isChecked = false;
     if (type === 'radio' || type === 'checkbox') {
       isChecked = (target as HTMLInputElement).checked;
     } else if (
-      maskInputOptions[
-        (target as Element).tagName.toLowerCase() as keyof MaskInputOptions
-      ] ||
-      maskInputOptions[type as keyof MaskInputOptions]
+      hasInputMaskOptions({
+        maskInputOptions,
+        maskInputSelector,
+        tagName: (target as HTMLElement).tagName,
+        type,
+      })
```

Comment:
Previous PR I had removed this condition, but that meant it was calling `maskInputValue` on every change, so `hasInputMaskOptions` is called before `maskInputValue`. Worst case it will get called twice (it's also called in `maskInputValue` to determine if it needs masking), but `hasInputMaskOptions` avoids calling `el.matches()/closest()` which is a bit costlier than the O(1) lookups in `hasInputMaskOptions`.

Reply:
That makes sense to me